core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -73,6 +74,16 @@ public static <T> CloseableIterable<T> filter(CloseableIterable<T> rows, Functio
     return filter.filter(rows);
   }
 
+  public static <T> CloseableIterable<T> marker(CloseableIterable<T> rows, Function<T, Long> rowToPosition,
+                                                PositionDeleteIndex deleteSet, Consumer<T> markDeleted) {
+    if (deleteSet.isEmpty()) {
+      return rows;
+    }
+
+    PositionSetDeleteMarker<T> deleteMarker = new PositionSetDeleteMarker<>(rowToPosition, deleteSet, markDeleted);
+    return deleteMarker.filter(rows);
```

Comment:
since `PositionSetDeleteMarker` always return true, we are not actually doing filter. it seems that we are mainly leverage the filter to traverse the iterable and get the side effect of calling `markDeleted` with matched rows. The semantics is a little odd to me. Maybe we don't need to introduce `PositionSetDeleteMarker` filter?

Reply:
Maybe just use this method from `CloseableIterable` to traverse? it returns the same object after applying the `Consumer` function.

```
  static <I, O> CloseableIterable<O> transform(CloseableIterable<I> iterable, Function<I, O> transform) {
```

Reply:
I think it traverses the rows and adds `is_deleted` value to the row.